clang/lib/Tooling/Tooling.cpp

```diff
@@ -899,6 +917,22 @@ int ClangTool::processFiles(llvm::StringRef File,bool &ProcessingFailed,
         CudaArgsAdjuster = combineAdjusters(
             std::move(CudaArgsAdjuster),
             getInsertArgumentAdjuster("-x", ArgumentInsertPosition::BEGIN));
+        std::string CUDAVerMajor =
+            "-D__CUDACC_VER_MAJOR__=" + std::to_string(SDKVersionMajor);
+        CudaArgsAdjuster = combineAdjusters(
+            std::move(CudaArgsAdjuster),
+            getInsertArgumentAdjuster(CUDAVerMajor.c_str(),
+                                      ArgumentInsertPosition::BEGIN));
+        std::string CUDAVerMinor =
+            "-D__CUDACC_VER_MINOR__=" + std::to_string(SDKVersionMinor);
+        CudaArgsAdjuster = combineAdjusters(
+            std::move(CudaArgsAdjuster),
+            getInsertArgumentAdjuster(CUDAVerMinor.c_str(),
+                                      ArgumentInsertPosition::BEGIN));
+        CudaArgsAdjuster = combineAdjusters(
+            std::move(CudaArgsAdjuster),
+            getInsertArgumentAdjuster("-fgpu-exclude-wrong-side-overloads",
+                                      ArgumentInsertPosition::BEGIN));
```

Comment:
Line920~Line935 is same as Line892~Line903, suggest summary the same logic into a function.